src/background/helpers.js

```diff
@@ -31,6 +33,22 @@ chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
     case 'openPrivateWindowWithUrl':
       chrome.windows.create({ url: msg.url, incognito: true });
       break;
+    case 'idle': {
```

Comment:
please don't call it `idle` or background idle as this reflects state of options observers only and only from one instance of time. The name have should have something about `options` in it.